### PR TITLE
Pronunciation assessment with stream for C++ language

### PR DESCRIPTION
**Pull request by jinshan1979**
## Purpose
 Pronunciation assessment with stream for C++ language
* ...

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

**Comment by jinshan1979** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: samples/cpp/windows/console/samples/speech_recognition_samples.cpp [hunk 1]
```diff
@@ -613,6 +615,68 @@ void PronunciationAssessmentWithMicrophone()
...
+
+    auto result = speechRecognizer->RecognizeOnceAsync().get();
+    if (result->Reason == ResultReason::Canceled)
+    {
+        auto cancellationDetails = CancellationDetails::FromResult(result);
+        std::cout << cancellationDetails->ErrorDetails << std::endl;
+    }
+    else
+    {
+        auto responseJson = result->Properties.GetProperty(PropertyId::SpeechServiceResponse_JsonResult, "");
```
Comment (yulin-li) [inline samples/cpp/windows/console/samples/speech_recognition_samples.cpp:640]: use `pronunciationAssessmentResult`

File: samples/cpp/windows/console/samples/speech_recognition_samples.cpp [hunk 2]
```diff
@@ -613,6 +615,68 @@ void PronunciationAssessmentWithMicrophone()
...
+
+    // Read audio data from file. In real scenario this can be from memory or network
+    std::ifstream file("whatstheweatherlike.wav", std::ios::binary | std::ios::ate);
+    auto audioDataWithHeaderSize = file.tellg();
+    file.seekg(46);
+    auto audioData = std::vector<uint8_t>(static_cast<size_t>(audioDataWithHeaderSize) - static_cast<size_t>(46));
+
+    file.read((char*)audioData.data(), audioData.size());
+
+    std::promise<int> resultReceived;
```
Comment (yulin-li) [inline samples/cpp/windows/console/samples/speech_recognition_samples.cpp:663]: without using it?

**Review by yulin-li — CHANGES_REQUESTED** [review]

File: samples/cpp/windows/console/samples/speech_recognition_samples.cpp (hunk 2 above)
Comment (jinshan1979) [reply samples/cpp/windows/console/samples/speech_recognition_samples.cpp:668]: adding resultReceived.get_future() and futureResult.get()

File: samples/cpp/windows/console/samples/speech_recognition_samples.cpp (hunk 1 above)
Comment (jinshan1979) [reply samples/cpp/windows/console/samples/speech_recognition_samples.cpp:640]: use the pronunciationAssessmentResult to get the scores related.

File: samples/cpp/windows/console/samples/samples.vcxproj [hunk 3]
```diff
@@ -107,11 +107,14 @@
       <PreprocessorDefinitions>_DEBUG;_CONSOLE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <ConformanceMode>true</ConformanceMode>
       <AdditionalIncludeDirectories>$(MSBuildThisFileDirectory)include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalUsingDirectories>
```
Comment (yulin-li) [inline samples/cpp/windows/console/samples/samples.vcxproj:110]: what's this for?

File: samples/cpp/windows/console/samples/speech_recognition_samples.cpp [hunk 4]
```diff
@@ -613,6 +615,75 @@ void PronunciationAssessmentWithMicrophone()
...
+    if (result->Reason == ResultReason::Canceled)
+    {
+        auto cancellationDetails = CancellationDetails::FromResult(result);
+        std::cout << cancellationDetails->ErrorDetails << std::endl;
+    }
+    else
+    {
+        auto responseJson = result->Properties.GetProperty(PropertyId::SpeechServiceResponse_JsonResult, "");
+        auto responsePA = PronunciationAssessmentResult::FromResult(result);
+        auto responseResult = "RECOGNIZED : Text = " + result->Text + "\n";
```
Comment (yulin-li) [inline samples/cpp/windows/console/samples/speech_recognition_samples.cpp:642]: we don't need this

File: samples/cpp/windows/console/samples/speech_recognition_samples.cpp [hunk 5]
```diff
@@ -613,6 +615,75 @@ void PronunciationAssessmentWithMicrophone()
...
+
+    auto result = speechRecognizer->RecognizeOnceAsync().get();
+    if (result->Reason == ResultReason::Canceled)
+    {
+        auto cancellationDetails = CancellationDetails::FromResult(result);
+        std::cout << cancellationDetails->ErrorDetails << std::endl;
+    }
+    else
+    {
+        auto responseJson = result->Properties.GetProperty(PropertyId::SpeechServiceResponse_JsonResult, "");
```
Comment (yulin-li) [inline samples/cpp/windows/console/samples/speech_recognition_samples.cpp:640]: remove

**Review by yulin-li — CHANGES_REQUESTED** [review]

File: samples/cpp/windows/console/samples/samples.vcxproj (hunk 3 above)
Comment (jinshan1979) [reply samples/cpp/windows/console/samples/samples.vcxproj:110]: 当时view代码的时候，虽然能编译，但是speech sdk不能自动使用识别，编辑过这个字段，后来又恢复到初始值（空）

File: samples/cpp/windows/console/samples/speech_recognition_samples.cpp (hunk 5 above)
Comment (jinshan1979) [reply samples/cpp/windows/console/samples/speech_recognition_samples.cpp]: removed

File: samples/cpp/windows/console/samples/speech_recognition_samples.cpp (hunk 4 above)
Comment (jinshan1979) [reply samples/cpp/windows/console/samples/speech_recognition_samples.cpp]: removed the result->Text

**Review by yulin-li — APPROVED** [review]

**Pull request merged**